pkg/libovsdb/schema.go

```diff
@@ -71,3 +79,344 @@ func (schema DatabaseSchema) validateOperations(operations ...Operation) bool {
+depends on internal, not directly marshallable fields. Therefore, in order to simplify the usage
+of this library, we define an ExtendedType that includes all possible column types (including
+atomic fields).
+*/
+
+//ExtendedType includes atomic types as defined in the RFC plus Enum, Map and Set
+type ExtendedType = string
+
+// RefType is used to define the possible RefTypes
+type RefType = string
+
+const (
+	//Unlimited is used to express unlimited "Max"
+	Unlimited int = -1
+
+	//Strong RefType
+	Strong RefType = "strong"
+	//Weak RefType
+	Weak RefType = "weak"
+
+	//ExtendedType associated with Atomic Types
+
+	//TypeInteger is equivalent to 'int'
+	TypeInteger ExtendedType = "integer"
+	//TypeReal is equivalent to 'float64'
+	TypeReal ExtendedType = "real"
+	//TypeBoolean is equivalent to 'bool'
+	TypeBoolean ExtendedType = "boolean"
+	//TypeString is equivalent to 'string'
+	TypeString ExtendedType = "string"
+	//TypeUUID is equivalent to 'libovsdb.UUID'
+	TypeUUID ExtendedType = "uuid"
+
+	//Extended Types used to summarize the interal type of the field.
+
+	//TypeEnum is an enumerator of type defined by Key.Type
+	TypeEnum ExtendedType = "enum"
+	//TypeMap is a map whose type depend on Key.Type and Value.Type
+	TypeMap ExtendedType = "map"
+	//TypeSet is a set whose type depend on Key.Type
+	TypeSet ExtendedType = "set"
+)
+
+// ColumnSchema is a column schema according to RFC7047
+type ColumnSchema struct {
+	// According to RFC7047, "type" field can be, either an <atomic-type>
+	// Or a ColumnTypeObject defined below. To try to simplify the usage, the
+	// json message will be parsed manually and Type will indicate the "extended"
+	// type. Depending on its value, more information may be available in TypeObj.
+	// E.g: If Type == TypeEnum, TypeObj.Key.Enum contains the possible values
+	Type      ExtendedType
+	TypeObj   *ColumnType
+	Ephemeral bool
+	Mutable   bool
+}
+
+// ColumnType is a type object as per RFC7047
+type ColumnType struct {
+	Key   *BaseType
+	Value *BaseType
+	Min   int
+	// Unlimited is expressed by the const value Unlimited (-1)
+	Max int
+}
+
+// BaseType is a base-type structure as per RFC7047
+type BaseType struct {
+	Type string `json:"type"`
+	// Enum will be parsed manually and set to a slice
+	// of possible values. They must be type-asserted to the
+	// corret type depending on the Type field
+	Enum       []interface{} `json:"_"`
+	MinReal    float64       `json:"minReal,omitempty"`
+	MaxReal    float64       `json:"maxReal,omitempty"`
+	MinInteger int           `json:"minInteger,omitempty"`
+	MaxInteger int           `json:"maxInteger,omitempty"`
+	MinLength  int           `json:"minLength,omitempty"`
+	MaxLength  int           `json:"maxLength,omitempty"`
+	RefTable   string        `json:"refTable,omitempty"`
+	RefType    RefType       `json:"refType,omitempty"`
+}
+
+// String returns a string representation of the (native) column type
```

Comment:
you create some kind of type encoding here, a comment with explanation can be useful